service/src/main/java/org/cbioportal/service/impl/ClinicalDataServiceImpl.java

```diff
@@ -15,103 +13,69 @@
+                pageSize, pageNumber, sortBy, direction);
     }
 
     @Override
     public BaseMeta getMetaSampleClinicalData(String studyId, String sampleId, String attributeId) {
 
-        return clinicalDataRepository.getMetaSampleClinicalData(Arrays.asList(studyId), Arrays.asList(sampleId),
-                attributeId);
+        return clinicalDataRepository.getMetaSampleClinicalData(studyId, sampleId, attributeId);
     }
 
     @Override
-    public List<PatientClinicalData> getAllClinicalDataOfPatientInStudy(String studyId, String patientId,
+    public List<ClinicalData> getAllClinicalDataOfPatientInStudy(String studyId, String patientId,
                                                                         String attributeId, String projection,
                                                                         Integer pageSize, Integer pageNumber,
                                                                         String sortBy, String direction) {
 
-        return clinicalDataRepository.getAllClinicalDataOfPatientInStudy(Arrays.asList(studyId),
-                Arrays.asList(patientId), attributeId, projection, pageSize, pageNumber, sortBy, direction);
+        return clinicalDataRepository.getAllClinicalDataOfPatientInStudy(studyId, patientId, attributeId, projection,
+                pageSize, pageNumber, sortBy, direction);
     }
 
     @Override
     public BaseMeta getMetaPatientClinicalData(String studyId, String patientId, String attributeId) {
 
-        return clinicalDataRepository.getMetaPatientClinicalData(Arrays.asList(studyId), Arrays.asList(patientId),
-                attributeId);
+        return clinicalDataRepository.getMetaPatientClinicalData(studyId, patientId, attributeId);
     }
 
     @Override
-    public List<? extends ClinicalDataSummary> getAllClinicalDataInStudy(String studyId, String attributeId,
+    public List<ClinicalData> getAllClinicalDataInStudy(String studyId, String attributeId,
                                                                          String clinicalDataType, String projection,
                                                                          Integer pageSize, Integer pageNumber,
                                                                          String sortBy, String direction) {
 
-        if (clinicalDataType.equals(SAMPLE_CLINICAL_DATA_TYPE)) {
-            return clinicalDataRepository.getAllClinicalDataOfSampleInStudy(Arrays.asList(studyId), null, attributeId,
-                    projection, pageSize, pageNumber, sortBy, direction);
-        } else {
-            return clinicalDataRepository.getAllClinicalDataOfPatientInStudy(Arrays.asList(studyId), null, attributeId,
-                    projection, pageSize, pageNumber, sortBy, direction);
-        }
+        return clinicalDataRepository.getAllClinicalDataInStudy(studyId, attributeId, clinicalDataType, projection,
```

Comment:
maybe here is a case where the "if" logic was eliminated at this level and pushed down into clinicalDataRepository. The alternative would be to push the "if" logic up to callers of the service --- making them choose between two different service level functions. It seems like there is not much shared code here between the two modes here because this is mostly a passthrough call.